docs/logging.rst

```diff
@@ -0,0 +1,169 @@
+.. _logging:
+
+Logging
+=======
+
+Flask uses standard Python :mod:`logging`. All Flask-related messages are
+logged under the ``'flask'`` logger namespace.
+:meth:`Flask.logger <flask.Flask.logger>` returns the logger named
+``'flask.app'``, and can be used to log messages for your application. ::
+
+    @app.route('/login', methods=['POST'])
+    def login():
+        user = get_user(request.form['username'])
+
+        if user.check_password(request.form['password']):
+            login_user(user)
+            app.logger.info('%s logged in successfully', user.username)
+            return redirect(url_for('index'))
+        else:
+            app.logger.info('%s failed to log in', user.username)
+            abort(401)
+
+
+Basic Configuration
+-------------------
+
+When you want to configure logging for your project, you should do it as soon
+as possible when the program starts. If :meth:`app.logger <flask.Flask.logger>`
+is accessed before logging is configured, it will add a default handler. If
+possible, configure logging before creating the application object.
+
+This example uses :func:`~logging.config.dictConfig` to create a logging
+configuration similar to Flask's default, except for all logs::
+
+    from logging.config import dictConfig
+
+    dictConfig({
+        'version': 1,
+        'formatters': {'default': {
+            'format': '[%(asctime)s] %(levelname)s in %(module)s: %(message)s',
+        }},
+        'handlers': {'wsgi': {
+            'class': 'logging.StreamHandler',
+            'stream': 'ext://flask.logging.wsgi_errors_stream',
+            'formatter': 'default'
+        }},
+        'root': {
+            'level': 'INFO',
+            'handlers': ['wsgi']
+        }
+    })
+
+    app = Flask(__name__)
+
+
+Default Configuration
+`````````````````````
+
+If you do not configure logging yourself, Flask will add a
+:class:`~logging.StreamHandler` automatically. During requests, it will write
+to the stream specified by the WSGI server in ``environ['wsgi.errors']``
+(which is usually :data:`sys.stderr`). Outside a request, it will log to
+:data:`sys.stderr`.
+
+
+Removing the Default Handler
+````````````````````````````
+
+If you configured logging after accessing
+:meth:`app.logger <flask.Flask.logger>`, and need to remove the default
+handler, you can import and remove it::
+
+    from flask.logging import default_handler
+
+    app.logger.removeHandler(default_handler)
```

Comment:
Will this code work if `app.logger` hasn't been accessed yet? In other words, can I run this code at any time and know that afterwards I've successfully opted out of Flask's logging?